src/core/search/indices.cc

```diff
@@ -67,6 +73,14 @@ void TextIndex::Add(DocId doc, string_view value) {
   }
 }
 
+void TextIndex::Remove(DocId doc, string_view value) {
+  for (const auto& word : GetWords(value)) {
+    auto& list = entries_[word];
+    if (auto it = lower_bound(list.begin(), list.end(), doc); it != list.end() && *it == doc)
```

Comment:
I personally find the usage of the if-initializer harder to read, but your call if you disagree
(also below)